src/notification.c

```diff
@@ -306,31 +320,85 @@ int notification_init(notification * n, int id)
+                                break;
+                        case 's':
+                                notification_replace_single_field(
+                                        &n->msg,
+                                        &substr,
+                                        n->summary,
+                                        n->markup);
+                                break;
+                        case 'b':
+                                notification_replace_single_field(
+                                        &n->msg,
+                                        &substr,
+                                        n->body,
+                                        n->markup);
+                                break;
+                        case 'I':
+                                notification_replace_single_field(
+                                        &n->msg,
+                                        &substr,
+                                        n->icon ? basename(n->icon) : "",
+                                        MARKUP_NO);
+                                break;
+                        case 'i':
+                                notification_replace_single_field(
+                                        &n->msg,
+                                        &substr,
+                                        n->icon ? n->icon : "",
+                                        MARKUP_NO);
+                                break;
+                        case 'p':
+                                if (n->progress)
+                                        sprintf(pg, "[%3d%%]", n->progress - 1);
+
+                                notification_replace_single_field(
+                                        &n->msg,
+                                        &substr,
+                                        n->progress ? pg : "",
+                                        MARKUP_NO);
+                                break;
+                        case 'n':
+                                if (n->progress)
+                                        sprintf(pg, "%d", n->progress - 1);
+
+                                notification_replace_single_field(
+                                        &n->msg,
+                                        &substr,
+                                        n->progress ? pg : "",
+                                        MARKUP_NO);
+                                break;
+                        case '%':
+                                notification_replace_single_field(
+                                        &n->msg,
+                                        &substr,
+                                        "%",
+                                        MARKUP_NO);
+                                break;
+                        default:
+                                fprintf(stderr, "WARNING: format_string %%%c"
+                                                " is unknown\n", substr[1]);
```

Comment:
There should be a different message if `substr[1]` is NULL, aka we're at the end of the string.

e.g. `WARNING: Trailing % mark in format`

Reply:
Treated.